api/src/main/java/org/infinispan/api/common/annotations/indexing/BasicProcessor.java

```diff
@@ -0,0 +1,29 @@
+package org.infinispan.api.common.annotations.indexing;
+
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.processing.PropertyMappingAnnotationProcessor;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.processing.PropertyMappingAnnotationProcessorContext;
+import org.hibernate.search.mapper.pojo.mapping.definition.programmatic.PropertyMappingGenericFieldOptionsStep;
+import org.hibernate.search.mapper.pojo.mapping.definition.programmatic.PropertyMappingStep;
+import org.infinispan.api.annotations.indexing.Basic;
+import org.infinispan.api.annotations.indexing.model.Values;
+
+public class BasicProcessor implements PropertyMappingAnnotationProcessor<Basic> {
+
+   @Override
+   public void process(PropertyMappingStep mapping, Basic annotation, PropertyMappingAnnotationProcessorContext context) {
```

Comment:
Could we move these `processors` to another module ? I don't like the fact that such code is delivered as part of an API module.

Reply:
yes the idea is to have these classes in a query module for instance